tests/std/tests/VSO_0157762_feature_test_macros/test.cpp

```diff
@@ -1513,6 +1513,20 @@ STATIC_ASSERT(__cpp_lib_bind_front == 201907L);
 #endif
 #endif
 
+#if CXX20_MODE && !defined(__EDG__) // TRANSITION, VSO-1041044

```

Comment:
Should that be `_HAS_CXX20 ?`

Reply:
That's a good question. This test is somewhat special; at the top it has:

https://github.com/microsoft/STL/blob/1cd4e071ff9834e0bc6ab6af4884840501aee906/tests/std/tests/VSO_0157762_feature_test_macros/test.cpp#L10-L32

This is because part of the test covers the MSVC compiler without including any STL headers, so we can't query `_HAS_CXX20`. (Arguably this should move into the compiler test suite; this is a relic of the fact that I implemented feature-test macros initially in both the compiler and library.) Later in the test, we include `<version>` but keep using the non-STL-internal macros for consistency.

So, I would say that this is intentional, although strange and perhaps we should consider refactoring this test. The argument for using `CXX20_MODE` here is that it's "independently derived" so it could possibly avoid problems like `_HAS_CXX20` being set weirdly, and that affecting the feature-test macros. I am inclined to leave this as-is (but then again I wrote it that way).

Reply:
it's also pre-existing and I don't wanna change all of the feature tests macro test in a pr to add bit_cast